starfish/stack.py

```diff
@@ -119,23 +119,26 @@ def __init__(self, image_partition: TileSet) -> None:
             c = tile.indices[Indices.CH]
             zlayer = tile.indices.get(Indices.Z, 0)
             data = tile.numpy_array
-            if max_size != data.dtype.itemsize:
-                if data.dtype.kind == "i" or data.dtype.kind == "u":
-                    # fixed point can be done with a simple multiply.
-                    src_range = np.iinfo(data.dtype).max - np.iinfo(data.dtype).min + 1
-                    dst_range = np.iinfo(self._data.dtype).max - np.iinfo(self._data.dtype).min + 1
-                    data = data * (dst_range / src_range)
-                warnings.warn(
-                    f"Tile "
-                    f"(R: {tile.indices[Indices.ROUND]} C: {tile.indices[Indices.CH]} "
-                    f"Z: {tile.indices[Indices.Z]}) has "
-                    f"dtype {data.dtype}.  One or more tiles is of a larger dtype "
-                    f"{self._data.dtype}.",
-                    DataFormatWarning)
+            data = img_as_float(data)
```

Comment:
What happens here if someone submits a uint16 and another uint32 tile? Probably worth a warning.

Reply:
Tony would put this back.